crates/nu-protocol/src/eval_const.rs

```diff
@@ -259,39 +263,51 @@ pub(crate) fn create_nu_constant(engine_state: &EngineState, span: Span) -> Valu
     // if not, if windows, use ALLUSERPROFILE\nushell\vendor\autoload
     // if not, if non-windows, if env var PREFIX is set, use PREFIX/share/nushell/vendor/autoload
     // if not, use the default /usr/share/nushell/vendor/autoload
 
-    // check to see if NU_VENDOR_AUTOLOAD_DIR env var is set, if not, use the default
-    Some(
-        option_env!("NU_VENDOR_AUTOLOAD_DIR")
-            .map(String::from)
-            .unwrap_or_else(|| {
-                if cfg!(windows) {
-                    let all_user_profile = match engine_state.get_env_var("ALLUSERPROFILE") {
-                        Some(v) => format!(
-                            "{}\\nushell\\vendor\\autoload",
-                            v.coerce_string().unwrap_or("C:\\ProgramData".into())
-                        ),
-                        None => "C:\\ProgramData\\nushell\\vendor\\autoload".into(),
-                    };
-                    all_user_profile
-                } else {
-                    // In non-Windows environments, if NU_VENDOR_AUTOLOAD_DIR is not set
-                    // check to see if PREFIX env var is set, and use it as PREFIX/nushell/vendor/autoload
-                    // otherwise default to /usr/share/nushell/vendor/autoload
-                    option_env!("PREFIX").map(String::from).map_or_else(
-                        || "/usr/local/share/nushell/vendor/autoload".into(),
-                        |prefix| format!("{}/share/nushell/vendor/autoload", prefix),
-                    )
-                }
-            })
-            .into(),
-    )
+    let into_autoload_path_fn = |mut path: PathBuf| {
+        path.push("nushell");
+        path.push("vendor");
+        path.push("autoload");
+        path
+    };
+    let mut dirs = Vec::new();
+
+    // platform base directies
+    // TODO: We're handling windows and not windows at the moment. using XDG_DATA_DIRS and falling back to `/usr/(local/)share` on MacOS might not be correct from my reading of the Apple developer documentation.
+    #[cfg(not(target_os = "windows"))]
+    // TODO: env var from engine state? How does that handle strings where a path is not valid unicode for some reason? Even if we do that here, the env vars will not be read from the engine state but the process in the `dirs_next` call below.
+    std::env::var_os("XDG_DATA_DIRS")
+        .unwrap_or_else(|| std::ffi::OsString::from("/usr/local/share/:/usr/share/"))
+        .as_encoded_bytes()
+        .split(|b| *b == b':')
+        .map(|split| {
+            let os_str = unsafe { std::ffi::OsStr::from_encoded_bytes_unchecked(split) };
+            into_autoload_path_fn(PathBuf::from(os_str))
+        })
+        .for_each(|path| dirs.push(path));
+    #[cfg(target_os = "windows")]
+    // TODO: this is different to the previous behaviour, but this would be correct IMO.
+    dirs_sys_next::known_folder(&winapi::um::knownfolders::FOLDERID_ProgramData)
```

Comment:
I'm not a fan of using winapi. We should use the window-rs crate if we want to do things this way.

Reply:
nushell already uses `dirs_next` and `dirs_sys_next`, which under the hood use winapi. Introducing `windows-rs` for just this single call would be blowing up the dependency tree quite a bit without much reason.

Reply:
Maybe we should use something more modern that dirs-next?

Reply:
I'm fine with also switching over `dirs_next` to `dirs`, which funnily enough is the more maintained version of these two, and then using the `windows-sys` crate instead of `winnapi`. But that's a bigger change.

Reply:
I've opened a PR for replacing dirs_next with dirs in https://github.com/nushell/nushell/pull/13384, rebased this on top of it and switched to using windows-rs.

If someone could test this, I'd appreciate it, but considering this is done, I'll be marking this thread as resolved either way.